### DISCUSSION TITLE
Launch asciinema when rofi ssh is launched

### DISCUSSION THREAD

**Discussion by cnekmp** [davatorium/rofi#1781] [Q&A]
I want to start recording via asciinema only when I ssh to remote host using rofi.

Via `.bashrc` I use following command that works as intended but `rofi` ignores `.bashrc` options

```
if [[ -z $ASCIINEMA_REC ]]; then
  mkdir -p ~/ShellHistory
  exec asciinema rec --append --quiet ~/ShellHistory/history_$(date +"%Y%m%d").cast
fi
```

I cannot to get it to work in `config.rasi` with changing to something like this:
```
ssh-command: "{terminal} -e  exec asciinema rec --append --quiet ~/ShellHistory/history_ssh_$(date +"%Y%m%d").cast && {terminal} -e ssh {host}";
 ```
Is there any way to achieve this?

My rofi configuration is:
```
configuration {
	modi: "combi";
	font: "Ubuntu Regular 10";
	show-icons: true;
	terminal: "alacritty";
	ssh-command: "{terminal} -e ssh {host}";
	combi-modi: "drun,ssh,window";
}
```

**Comment by DaveDavenport** [comment]
rofi does nothing with your bashrc (its not a shell).  Rofi launches in the environment whatever it was for the tool launching rofi.

Also this:

```bash
~/ShellHistory/history_ssh_$(date +"%Y%m%d").cast
```
won't be expanded, it will be passed as argument to the application as-is.

To fix this, i see two solutions:
 - Make sure the tool launching rofi has the right environment and don't use shell expansion/replacement.
 - Make a small shell wrapper script and have rofi call that script.
 - Make rofi execute bash that then executes your commandline and let it do all the fancy shell stuff.

**Comment by cnekmp** [comment]
I got your point, I'm launching Rofi via qtile:

`Key([mod], "r", lazy.spawn("rofi -show combi"), desc="Spawn Rofi"),`

Yet I'm thinking how can I manage to work it together, because I need ssh option of Rofi to scan known_hosts (I don't want to write wrapper script that will parse it). Anyway, thanks for the help.

**Reply by DaveDavenport** [reply]
? I don't understand that.
Just make rofi (via ssh-command) call your wrapper that then does everything you try to do in the -ssh-command before.
You still use rofi, but you then have a little shell script where you can launch your recording and everything.

Something like (not checked at all):

```bash
#!/usr/bin/env bash
asciinema rec --append --quiet ~/ShellHistory/history_ssh_$(date +"%Y%m%d").cast &
ssh $@
```

and you launch this via:
```css
ssh-command: "{terminal} -e myscript.sh {host}
```

**Reply by cnekmp** [reply]
Yeah, I've already tried that method. It works but not as intended: it doesn't record anything. Tried with `script` command (even maunaly via terminal)  - no success. 
`script` command outputs to file as:

`Script started on 2023-01-16 17:20:38+04:00 [<not executed on terminal>]`

This was my script that I was running. Turns out, it has nothing to do with `rofi`. Will look further for solution and report back if I will find one.
Closing this case, thanks a lot for your support @DaveDavenport 

P.S. Script that I was using for tests:

```
#!/bin/bash
request=$*

echo "$(date) triggered $request" >> /home/myuser/my.log &
script -af &
asciinema rec --append ~/ShellHistory/history_ssh_$(date +"%Y%m%d").cast &

ssh "$request"
```
